### ISSUE TITLE
methods in random module fail when arrays with associative domain are passed

### ISSUE THREAD

**Issue by CaptainSharf** [chapel-lang/chapel#16425]
### Summary of Problem
Invoking the methods `fillRandom`,`shuffle` and `permutation` on arrays with associative domains gives vague error. Should these methods include compile/runtime error handling, when arrays with non rectangular domains are passed as arguments?

### Steps to Reproduce

**Source Code:**

```chapel
use Random;
var dom = {1,2,3};
var arr:[dom] real;
fillRandom(arr);
```

**Compile command:**
`chpl random_test.chpl`

**Output**
```
$CHPL_HOME/modules/dists/DSIUtil.chpl:218: In function 'computeZeroBasedDomain':
$CHPL_HOME/modules/dists/DSIUtil.chpl:219: error: unresolved call 'unmanaged DefaultAssociativeDom(int(64),true).dsiDims()'
$CHPL_HOME/modules/internal/DefaultRectangular.chpl:554: note: this candidate did not match: DefaultRectangularDom.dsiDims()
$CHPL_HOME/modules/internal/ChapelArray.chpl:1503: note: because method call receiver with type unmanaged DefaultAssociativeDom(int(64),true)
$CHPL_HOME/modules/internal/DefaultRectangular.chpl:146: note: is passed to formal 'this: domain'
$CHPL_HOME/modules/internal/ChapelDistribution.chpl:602: note: candidates are: BaseSparseDom.dsiDims()
$CHPL_HOME/modules/standard/Random.chpl:1583: Function 'computeZeroBasedDomain' instantiated as: computeZeroBasedDomain(dom: DefaultAssociativeDom(int(64),true))
```

### Configuration Information
- `chpl --version`: chpl version 1.22.1
- `gcc --version`: Apple LLVM version 10.0.1 (clang-1001.0.46.4)

**Comment by bradcray** [comment]
It would be fairly easy to create a compile-time check to prevent these cases from occurring (if you're interested in pursuing that and you need more information, let me know).  But I could also imagine a user wanting to be able to permute , shuffle, or fillRandom on non-rectangular arrays.  A third approach could be to generate a nicer compiler error for now and mark them as a feature request or unimplemented feature.

**Comment by CaptainSharf** [comment]
Thanks for the feedback @bradcray , I will work on this

**Comment by CaptainSharf** [comment]
Fixed this issue as part of the PR #16441
